### PR TITLE
MMCA 4226

### PR DESCRIPTION
**Pull request by sameerhmrc**
JIRA Link : https://jira.tools.tax.service.gov.uk/browse/MMCA-4226

### CODE REVIEW

File: test/controllers/AuthorizedToViewControllerSpec.scala [hunk 1]
```diff
@@ -498,8 +498,33 @@ class AuthorizedToViewControllerSpec extends SpecBase {
...
+          messages(app)("cf.authorities.notification-panel.a.gb-authority")
+        html.getElementById("gb-csv-authority-link").attr("href") mustBe gbStanAuthFile154Url
+
+        html.text().contains("You cannot search your own EORI number") shouldBe true
+      }
+    }
+
     "Display error message if searching your own account number" in new Setup {
       when(mockDataStoreService.getXiEori(any)(any)).thenReturn(Future.successful(None))
+      when(mockApiService.getAccounts(ArgumentMatchers.eq(newUser().eori))(any)).thenReturn(Future.successful(cdsAccounts))
```
Comment (tyagikumarvineet) [inline test/controllers/AuthorizedToViewControllerSpec.scala:526]: Could you please split (better to do after thenReturn) the line over multiple line as its very long line? Same comment for line no 527, 543, 544, 641 and 642

File: app/controllers/AuthorizedToViewController.scala [hunk 2]
```diff
@@ -104,11 +104,15 @@ class AuthorizedToViewController @Inject()(authenticate: IdentifierAction,
     val searchQuery = stripWithWhitespace(query)
 
     val result = for {
-      cdsAccounts: CDSAccounts <- apiService.getAccounts(request.user.eori)
+      gbEoriAccounts: CDSAccounts <- apiService.getAccounts(request.user.eori)
       xiEORI: Option[EORI] <- dataStoreService.getXiEori(request.user.eori)
+
```
Comment (tyagikumarvineet) [inline app/controllers/AuthorizedToViewController.scala:109]: Could you please remove this line break to improve readability inside for yield?
Same for line no 111

Comment (sameerhmrc) [reply app/controllers/AuthorizedToViewController.scala]: Code chane done

File: app/controllers/AuthorizedToViewController.scala [hunk 3]
```diff
@@ -104,11 +104,17 @@ class AuthorizedToViewController @Inject()(authenticate: IdentifierAction,
...
+      xiEoriAccounts: CDSAccounts <- xiEORI match {
```
[the remark this answers is not in the document]
Comment (sameerhmrc) [reply app/controllers/AuthorizedToViewController.scala]: Code change done

**Pull request merged**